packages/vite/src/node/plugins/html.ts

```diff
@@ -121,14 +121,261 @@ export function addToHTMLProxyTransformResult(
   htmlProxyResult.set(hash, code)
 }
 
-// this extends the config in @vue/compiler-sfc with <link href>
-export const assetAttrsConfig: Record<string, string[]> = {
-  link: ['href'],
-  video: ['src', 'poster'],
-  source: ['src', 'srcset'],
-  img: ['src', 'srcset'],
-  image: ['xlink:href', 'href'],
-  use: ['xlink:href', 'href']
+export interface HtmlAssetSource {
+  tag: string
+  attributes: string[]
+  type: 'src' | 'srcset'
+  filter?: (attribute: string, value: string, node: ElementNode) => boolean
+}
+
+interface AttributeNodeContext {
+  context: {
+    sourceType: 'src' | 'srcset'
+  }
+}
+
+export const defaultHtmlAssetSources: HtmlAssetSource[] = [
```

Comment:
I think it would be great to extract this and it's related types/functions as it's own file as this is HUGE. Maybe we can put it in the `src/node` folder